src/freenet/client/async/SplitFileInserterStorage.java

```diff
@@ -218,7 +223,16 @@
 
     private final int overallStatusLength;
     
+    /** Protects cooldown calculations and noBlocksToSend. You MUST NOT take this lock while 
+     * holding any other locks!!!
```

Comment:
I'm not familiar enough with the code to have an opinion yet. How much more memory usage? I don't get the impression that Fred itself is particularly memory-hungry - it always seems to be plugins or a huge queue causing problems. What complaints are you referring to? Are there bugs filed? Can we change to separate getters for each segment while maintaining API backwards compatibility? If not then I'm not interested in the approach.